assets/data-port/import/data/selectors.js

```diff
@@ -124,13 +124,32 @@ export const getSuccessResults = ( { done } ) =>
 /**
  * Get logs by severity.
  *
- * @param {Object} state       Current state.
- * @param {Object} state.done  The object which contains the logs of the job.
+ * @param {Object} state        Current state.
+ * @param {Object} state.done   The object which contains the logs of the job.
+ * @param {Object} state.upload The object which contains the uploads.
  *
  * @return {Object} Object with the logs by severity.
  */
-export const getLogsBySeverity = ( { done } ) =>
-	groupBy( get( done, 'logs.items', [] ), 'severity' );
+export const getLogsBySeverity = ( { done, upload } ) => {
+	const sortObject = DONE_KEYS.reduce(
+		( acc, cur, index ) => ( {
+			...acc,
+			[ cur ]: index,
+		} ),
+		{}
+	);
+
+	const items = get( done, 'logs.items', [] )
+		// Add filename to the results.
+		.map( ( i ) => ( {
+			...i,
+			filename: get( upload, i.type + 's.filename', '' ),
+		} ) )
+		// Sort the results by type.
+		.sort( ( a, b ) => sortObject[ a.type ] - sortObject[ b.type ] );
```

Comment:
Might be simpler without the sortObject:
`(a,b) => DONE_KEYS.indexOf(a.type) - DONE_KEYS.indexOf(b.type)`
Not critical :)

Reply:
Actually I did like that to be more performative in case of big logs. So we don't need to make 2 `indexOf` each loop of the sort. 😊